### PR TITLE
Feature/get folder items

### PR DESCRIPTION
**Pull request by valer-kovac**
Updated GetFolderItemsAsync and GetFolderItems_ValidResponse_ValidFolder UT

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @valer-kovac has signed the CLA. Thanks for the pull request!

File: Box.V2/Extensions/BoxExtensions.cs [hunk 1]
```diff
@@ -11,26 +11,47 @@ internal static class BoxExtensions
...
         /// </summary>
         /// <typeparam name="T">Type of the object being checked</typeparam>
-        /// <param name="param"></param>
-        /// <param name="name"></param>
+        /// <param name="param">Object being checked</param>
+        /// <param name="name">Object name</param>
         internal static T ThrowIfNull<T>(this T param, string name) where T : class
         {
             if (param == null)
+            {
```
Comment (coolboy) [inline Box.V2/Extensions/BoxExtensions.cs]: Thanks for adding the {}

File: Box.V2/Managers/BoxFoldersManager.cs [hunk 2]
```diff
@@ -37,16 +37,27 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
...
-        public async Task<BoxCollection<BoxItem>> GetFolderItemsAsync(string id, int limit, int offset = 0, List<string> fields = null)
+        /// <param name="id">The folder id</param>
+        /// <param name="limit">The maximum number of items to return in a page. The default is 100 and the max is 1000.</param>
+        /// <param name="offset">The offset at which to begin the response. An offset of value of 0 will start at the beginning of 
+        /// the folder-listing. Note: If there are hidden items in your previous response, your next offset should be = offset + limit,
+        /// not the # of records you received back. The default is 0.</param>
+        /// <param name="fields">Attribute(s) to include in the response</param>
+        /// <returns>A collection of items contained in the folder is returned. An error is thrown if the folder does not exist,
+        /// or if any of the parameters are invalid.</returns>
+        public async Task<BoxCollection<BoxItem>> GetFolderItemsAsync(string id, int limit=100, int offset = 0, List<string> fields = null)
```
Comment (coolboy) [inline Box.V2/Managers/BoxFoldersManager.cs]: limit = 100

File: Box.V2/Managers/BoxFoldersManager.cs [hunk 3]
```diff
@@ -37,16 +37,27 @@ public BoxFoldersManager(IBoxConfig config, IBoxService service, IBoxConverter c
...
         {
             id.ThrowIfNullOrWhiteSpace("id");
+            limit.ThrowIfHigherThan(1000, "limit");
```
Comment (coolboy) [inline Box.V2/Managers/BoxFoldersManager.cs]: make 1000 into a constant or config

**Comment by valer-kovac** [comment]
Incorporated review comments

**Comment by coolboy** [comment]
@valer-kovac conflict

**Comment by marek-vysoky** [comment]
@coolboy  Conflict resolved

**Comment by cburnette** [comment]
@coolboy Needs review

File: Box.V2.Test/BoxResourceManagerTest.cs [hunk 4]
```diff
@@ -42,7 +44,8 @@ public BoxResourceManagerTest()
 
             _authRepository = new AuthRepository(_config.Object, _service, _converter, new OAuthSession("fakeAccessToken", "fakeRefreshToken", 3600, "bearer"));
         }
-        public static T CreateInstanceNonPublicConstructor<T>()
+
+		public static T CreateInstanceNonPublicConstructor<T>()
```
Comment (coolboy) [inline Box.V2.Test/BoxResourceManagerTest.cs]: spacing

File: Box.V2/Extensions/BoxExtensions.cs [hunk 5]
```diff
@@ -11,28 +11,58 @@ internal static class BoxExtensions
         /// Checks if the object is null 
         /// </summary>
         /// <typeparam name="T">Type of the object being checked</typeparam>
-        /// <param name="param"></param>
-        /// <param name="name"></param>
-        internal static T ThrowIfNull<T>(this T param, string name) 
+        /// <param name="param">Object being checked</param>
+        /// <param name="name">Object name</param>
+		/// <returns>Param object if valid</returns>
```
Comment (coolboy) [inline Box.V2/Extensions/BoxExtensions.cs]: spacing

File: Box.V2/Managers/BoxMetadataManager.cs [hunk 6]
```diff
@@ -173,7 +173,7 @@ public BoxMetadataManager(IBoxConfig config, IBoxService service, IBoxConverter
             foreach (BoxMetadataUpdate update in updates)
             {
                 update.Path.ThrowIfNullOrWhiteSpace("Path");
-                update.Op.ThrowIfNull("Op");
+                update.Op.ThrowIfNullEnum("Op");
```
Comment (coolboy) [inline Box.V2/Managers/BoxMetadataManager.cs]: check my comment on #190

**Comment by coolboy** [comment]
@cburnette done reviewing, @marek-vysoky please resolve the conflict and address my comments

**Comment by marek-vysoky** [comment]
@coolboy can you check again?

File: Box.V2.Test/BoxFilesManagerTest.cs [hunk 7]
```diff
@@ -59,27 +59,45 @@ public BoxFilesManagerTest()
...
-                }));
+                }))
+                .Callback<IBoxRequest>(r => boxRequest = r as BoxMultiPartRequest);
 
             var fakeFileRequest = new BoxFileRequest()
             {
                 Name = "test.txt",
-                ContentCreatedAt = DateTime.Now,
-                ContentModifiedAt = DateTime.Now,
+                ContentCreatedAt = new DateTime(2016, 8, 27),
```
Comment (coolboy) [inline Box.V2.Test/BoxFilesManagerTest.cs]: timezone specific code

File: Box.V2.Test/BoxResourceManagerTest.cs [hunk 8]
```diff
@@ -57,5 +74,14 @@ public static T CreateInstanceNonPublicConstructor<T>()
                                System.Threading.Thread.CurrentThread.CurrentCulture);
             return inst;
         }
+
+        public static bool AreJsonStringsEqual(string sourceJsonString, string targetJsonString)
```
Comment (coolboy) [inline Box.V2.Test/BoxResourceManagerTest.cs]: try to have the latest code

**Comment by marek-vysoky** [comment]
@coolboy  I've changed validation of datetime ContentCreatedAt, ContentModifiedAt, please can you review it?

**Comment by cburnette** [comment]
functionality exists in master

**Pull request closed without merging**